embeddings/data/datamodule.py

```diff
@@ -124,38 +136,45 @@ def downsample_dataset(self) -> None:
                 )
                 self.dataset[column_name] = downsampled_data["test"]
 
-    def prepare_data(self) -> None:
-        self.load_dataset(preparation_step=True)
-        AutoTokenizer.from_pretrained(self.tokenizer_name_or_path)
-
-    def setup(self, stage: Optional[str] = None) -> None:
-        self.dataset = self.load_dataset()
-        self.downsample_dataset()
-        self.num_classes = self.get_num_classes()
-        self.process_data()
-
     def process_data(self) -> None:
         columns = [c for c in self.dataset["train"].column_names if c not in self.LOADER_COLUMNS]
         self.dataset = self.dataset.map(
             self.convert_to_features,
             batched=True,
+            batch_size=self.processing_batch_size,
             remove_columns=columns,
         )
+        self._class_encode_column("labels")
         self.dataset.set_format(type="torch")
 
     def train_dataloader(self) -> DataLoader[HuggingFaceDataset]:
-        return DataLoader(self.dataset["train"], batch_size=self.train_batch_size, shuffle=True)  # type: ignore
+        return DataLoader(
+            dataset=self.dataset["train"],  # type: ignore
+            batch_size=self.train_batch_size,
+            collate_fn=self.collate_fn,
+            shuffle=True,
+        )
 
     # Ignoring the type of val_dataloader method from supertype "DataHooks" allowing for None
     # and training without validation dataset.
     def val_dataloader(self) -> Optional[DataLoader[HuggingFaceDataset]]:  # type: ignore
         if "validation" in self.dataset:
-            return DataLoader(self.dataset["validation"], batch_size=self.eval_batch_size, shuffle=False)  # type: ignore
+            return DataLoader(
+                dataset=self.dataset["validation"],  # type: ignore
+                batch_size=self.eval_batch_size,
+                collate_fn=self.collate_fn,
+                shuffle=True,
+            )
         else:
             return None
 
     def test_dataloader(self) -> DataLoader[HuggingFaceDataset]:
-        return DataLoader(self.dataset["test"], batch_size=self.eval_batch_size, shuffle=False)  # type: ignore
+        return DataLoader(
+            dataset=self.dataset["test"],  # type: ignore
+            batch_size=self.eval_batch_size,
+            collate_fn=self.collate_fn,
+            shuffle=True,
```

Comment:
Test dataloader shouldn't be shuffled
```suggestion
            shuffle=False,
```